spec/order_spec.rb

```diff
@@ -0,0 +1,107 @@
+require "order"
+
+describe Order do
+  let(:dish) { double(:dish, name: "burger", price: 7) }
+  let(:restaurant) { double(:restaurant, menu: [dish], name: "mcdonalds") }
+  let(:SMS) { double(:SMS) }
+  let(:sms) { double(:sms) }
+  subject(:order) { described_class.new(restaurant) }
+
+  describe "#initialize" do
+    it "ensures it gets initialized with a total =0" do
+      expect(subject.total).to eq(0)
+    end
+
+    it "ensures it gets initialized with no items" do
+      expect(subject.items).to eq([])
+    end
+
+    it "ensures it gets initialized with a restaurant" do
+      expect(subject.restaurant).to eq(restaurant)
+    end
+
+    it "ensures a new order gets initialized with an id number" do
+      expect(subject.id).to be_kind_of(Numeric)
+    end
+
+    order1 = Order.new("restaurant1")
+    order2 = Order.new("restaurant2")
+    order3 = Order.new("restaurant3")
+    order4 = Order.new("restaurant4")
+
+    [order2.id - order1.id, order3.id - order2.id, order4.id - order3.id].each do |id_difference|
```

Comment:
This is a little hard to read. Generally avoid doing things which are too 'neat' in a test - the emphasis should be on readability.